browser/registry_test.go

```diff
@@ -296,6 +296,48 @@ func TestBrowserRegistry(t *testing.T) {
 
 		assert.True(t, browserRegistry.stopped.Load())
 	})
+
+	// This test ensures that the chromium browser's lifecycle is not controlled
+	// by the vu context.
+	t.Run("dont_close_browser_on_vu_context_close", func(t *testing.T) {
```

Comment:
This test will work with main too, but it's good to have still.

What is missing is an actual test to see what happens when SIGTERM is sent. Do you think it's a good idea to create a test using bash to:

1. Run a blocking test
2. run `kill -15 {PID}`
3. Ensure the test aborts soonish (within 10secs)

Is there a way to do this within the go code?

Reply:
Yes, we can do this with Go as we've similarly done before here: https://github.com/grafana/xk6-browser/blob/cc94235cae6259435d234b371264055510526308/tests/browser_test.go#L286-L324

Reply:
Ah, so i need to perform a SIGTERM on the k6 process, not on chromium.